packages/orchestrator/internal/server/sandboxes.go

```diff
@@ -214,6 +274,31 @@ func (s *server) Delete(ctxConn context.Context, in *orchestrator.SandboxDeleteR
 		}
 	}()
 
+	go func() {
+		sandboxLifeCycleEventsWriteFlag, flagErr := s.featureFlags.BoolFlag(
+			featureflags.SandboxLifeCycleEventsWriteFlagName, sbx.Config.SandboxId)
+		if flagErr != nil {
+			zap.L().Error("soft failing during sandbox lifecycle events write feature flag receive", zap.Error(flagErr))
+			return
+		}
+
+		if sandboxLifeCycleEventsWriteFlag {
+			err := s.clickhouseClient.InsertSandboxEvent(context.Background(), clickhouse.SandboxEvent{
+				Timestamp:          time.Now().UTC(),
+				SandboxID:          sbx.Config.SandboxId,
+				SandboxTemplateID:  sbx.Config.TemplateId,
+				SandboxBuildID:     sbx.Config.BuildId,
+				SandboxTeamID:      sbx.Config.TeamId,
+				SandboxExecutionID: sbx.Config.ExecutionId,
+				EventCategory:      string(clickhouse.SandboxEventCategoryLifecycle),
+				EventLabel:         string(clickhouse.SandboxEventLabelKill),
+			})
+			if err != nil {
+				sbxlogger.I(sbx).Error("error inserting sandbox event during kill", zap.Error(err))
+			}
+		}
+	}()
```

Comment:
I would do the check outside of go routine, if the check is false you don't need to create go routine which is more intensive then bool check (which you will do anyway)